simulator/src/data/undo.js

```diff
@@ -9,25 +9,29 @@ import Scope, { scopeList } from '../circuit';
 import { loadScope } from './load';
 import { updateRestrictedElementsInScope } from '../restrictedElementDiv';
 import { forceResetNodesSet } from '../engine';
-
+import { scheduleBackup } from './backupCircuit';
 /**
  * Function called to generate a prompt to save an image
  * @param {Scope=} - the circuit in which we want to call undo
  * @category data
  * @exports undo
  */
 export default function undo(scope = globalScope) {
+    scheduleBackup(); // Backup current globalscope
```

Comment:
I see why this is a good idea, but this is actually causing some problems. Sometimes, the circuit data is different so the scheduleBackup adds it to the backups array although the circuit corresponding to the data is the same. This then stalls the undos because everytime you undo, a new data is added and it is then popped, this keeps the backups array same forever. 
So I think it's best to remove this line

Reply:
But this line is important, without `scheduleBackup()` the functionality of undo and redo function will be disturbed.
Have a look at this, 

https://user-images.githubusercontent.com/76155456/145218888-a1984ae8-8a6b-4bf9-a152-3717eb2761ad.mov

without `scheduleBackup()` we would not able to redo the last work done by the user. 
We can go for some alternative.
I’m eager to receive your feedback.

Reply:
From what I understand, the reason that line is important is because we need to make sure that right before we undo, the top state of backups array is the current state. Without this it's possible that you pop a previous state that techincally leaves you 2 states behind in one undo (the final state is now unreachable). Right?

Then we might need to figure out some other way to achive that

Reply:
Yes, I meant to say the same.

Reply:
I guess you can remove that line for now and we can merge this, we can think about a better solution in another PR. Something is better than nothing. What do you say?

Reply:
Yes right @gr455, I will make a new commit after removing this line...